include/service/entities/article_service.js

```diff
@@ -435,88 +461,150 @@ module.exports = function ArticleServiceModule(pb) {
             });
         });
     };
-
+    
     /**
-     * Retrieves the meta info for an article or page
-     *
+     * Retrieves the SEO metadata for the specified content.  
      * @method getMetaInfo
-     * @param {Object}   article An article or page object
-     * @param {Function} cb      Callback function
+     * @param {Object} article The article to retrieve information for
+     * @param {Function} cb A callback that takes two parameters.  The first is 
+     * an Error, if occurred.  The second is an object that contains 4 
+     * properties: 
+     * title - the SEO title, 
+     * description - the SEO description, 
+     * keywords - an array of SEO keywords that describe the content, 
+     * thumbnail - a URI path to the thumbnail image 
      */
-    ArticleService.getMetaInfo = function(article, cb)
-    {
-        if(typeof article === 'undefined')
-        {
-            cb('', '', '');
-            return;
+    ArticleService.prototype.getMetaInfo = function(article, cb) {
```

Comment:
@markbaird This is where the issue was.  You were correct that things were not playing nicely when called in parallel.  I ended up deprecated the function directly off of the prototype and created a prototype function that in turn tasks itself out and executes in parallel.  

The whole exercise helped me find a few performance issues so a big thanks for the find.